pkg/database/sql.go

```diff
@@ -325,7 +369,14 @@ func (d *db) SQLQueryPrepared(stmt *sql.SelectStmt, namedParams []*schema.NamedP
 	cols := make([]*schema.Column, len(colDescriptors))
 
 	for i, c := range colDescriptors {
-		cols[i] = &schema.Column{Name: c.Selector, Type: c.Type}
+		des := &sql.ColDescriptor{
```

Comment:
Why it is now needed to rewrite those selectors?

Reply:
good point, the sql catalog and data are stored in the same store so to simplify replication and state validation.  So in order to support replicas with a different name than the source database, there must be an isolation from database name and internal database name, the solution included in this PR is the following: databases receive a fixed name `dbInstance` as there is a single database created in the embedded storage, but in order to return result sets with a different database name, selectors needs to provide a structured data type for its manipulation without incurring in string interpretation

Reply:
OK, was not aware that the replication can be done to a db with a different name than the original (or did I understand this incorrectly and there's some special use-case where the replication needs to rename DBs?).

If only the db name needs to change, why not:

```suggestion
		des := *c
		des.Database = d.options.dbName
```
?

That way any future modifications to the `ColDescriptor` struct will be much easier.